server/pkg/api/cluster_apps.go

```diff
@@ -120,3 +120,56 @@ func (s *Server) GetClusterApp(ctx context.Context, request *serverpb.GetCluster
 	*serverpb.GetClusterAppResponse, error) {
 	return &serverpb.GetClusterAppResponse{}, nil
 }
+
+func (s *Server) configureSSOForApp(ctx context.Context, orgId, clusterID string, app *agentpb.AppData) error {
+
+	a, err := s.agentHandeler.GetAgent(orgId, clusterID)
+	if err != nil {
+		s.log.Error("failed to connect to agent", err)
+		return err
+	}
```

Comment:
agent is already fetched in the registration, same can be reused passing the agent